src/Executables/Auth/PacketHandlers/LoginRequestHandler.cs

```diff
@@ -59,10 +82,19 @@ public async Task ExecuteAsync(AuthPacketContext<LoginRequest> ctx, Cancellation
         }
         catch (Exception e)
         {
-            _logger.LogWarning("Failed to verify password for account {Username}: {Message}", ctx.Packet.Username, e.Message);
-            status = "WRONGPWD";
+            _logger.LogWarning("Failed to verify password for account {Username}: {Message}", ctx.Packet.Username,
+                e.Message);
+            status = LoginFailedBecause.WrongPwd.ToString().ToUpper();
         }
 
+        // Check if the account is already logged in
+        var isAlreadyLoggedIn = await CheckExistingConnectionOf(account);
+        if (isAlreadyLoggedIn)
+        {
+            status = LoginFailedBecause.Already.ToString().ToUpper();
```

Comment:
Here, we could inform the connected client via an info message or we could allow the login on second (or third) attempt by just removing the key from redis and closing existing connection

Reply:
iirc the original implementation tried to log out the currently connected player once this login attempt fails. Maybe we should try reimplementing that?

Reply:
I agree we should. There should  some kind of flagging/logging the recurring attempts for possible attacks. Will do!

Reply:
you can take a look at SessionManager for mentioned behavior 👍🏻